test/contracts/LLL_ENS.cpp

```diff
@@ -0,0 +1,499 @@
+
+          ;; Nothing to return.
+          (stop)))
+
+      ;; ----------------------------------------------------------------------
+      ;; @notice Sets the resolver address for the specified node.
+      ;; @dev Signature: setResolver(bytes32,address)
+      ;; @param node The node to update.
+      ;; @param new-resolver The address of the resolver.
+
+      (def 'node (calldataload 0x04))
+      (def 'new-resolver (calldataload 0x24))
+
+      (function set-node-resolver
+        (seq (only-node-owner node)
+
+          ;; Transfer ownership by storing passed-in address.
+          (set-resolver node new-resolver)
+
+          ;; Emit an event about the change of resolver.
+          ;; NewResolver(bytes32 indexed node, address resolver);
+          (mstore call-result new-resolver)
+          (log2 call-result 32
+              (sha3 0x00 (lit 0x00 "NewResolver(bytes32,address)")) node)
+
+          ;; Nothing to return.
+          (stop)))
+
+      ;; ----------------------------------------------------------------------
+      ;; @notice Sets the TTL for the specified node.
+      ;; @dev Signature: setTTL(bytes32,uint64)
+      ;; @param node The node to update.
+      ;; @param ttl The TTL in seconds.
+
+      (def 'node (calldataload 0x04))
+      (def 'new-ttl (calldataload 0x24))
+
+      (function set-node-ttl
+        (seq (only-node-owner node)
+
+          ;; Set new TTL by storing passed-in time.
+          (set-ttl node new-ttl)
+
+          ;; Emit an event about the change of TTL.
+          ;; NewTTL(bytes32 indexed node, uint64 ttl);
+          (mstore call-result new-ttl)
+          (log2 call-result 32
+              (sha3 0x00 (lit 0x00 "NewTTL(bytes32,uint64)")) node)
+
+          ;; Nothing to return.
+          (stop)))
+
+      ;; ----------------------------------------------------------------------
+      ;; @notice Fallback: No functions matched the function ID provided.
+
+      (jump invalid-location)))
+
+)
+)DELIMITER";
+
+static unique_ptr<bytes> s_compiledEns;
+
+class LLLENSTestFramework: public LLLExecutionFramework
+{
+protected:
+	void deployEns()
+	{
+		if (!s_compiledEns)
+		{
+			vector<string> errors;
+			s_compiledEns.reset(new bytes(compileLLL(ensCode, false, &errors)));
+			BOOST_REQUIRE(errors.empty());
+		}
+		sendMessage(*s_compiledEns, true);
+		BOOST_REQUIRE(!m_output.empty());
+	}
+
+};
+
+}
+
+// Test suite for the deployed ENS Registry implementation written in LLL
+BOOST_FIXTURE_TEST_SUITE(LLLENS, LLLENSTestFramework)
+
+BOOST_AUTO_TEST_CASE(creation)
+{
+	deployEns();
+
+	// Root node 0x00 should initially be owned by the deploying account, account(0).
+	BOOST_CHECK(callContractFunction("owner(bytes32)", 0x00) == encodeArgs(ACCOUNT(0)));
+}
+
+BOOST_AUTO_TEST_CASE(transfer_ownership)
+{
+	deployEns();
+
+	// Transfer ownership of root node from account(0) to account(1).
+	BOOST_REQUIRE(callContractFunction("setOwner(bytes32,address)", 0x00, ACCOUNT(1)) == SILENT);
```

Comment:
Please use explicit `encodeArgs()` instead of `SILENT` in the entire file. It is more readable to someone familiar with the tests.